server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -179,37 +178,33 @@ void start() {
                 final DiscoveryNode node = nodes[i];
                 final String nodeId = node.getId();
                 try {
-                    if (node == null) {
```

Comment:
reason: `node` is used before, cannot be null here

Reply:
I wonder if that use is actually a bug....

Reply:
Good point. That use is `nodeId = node.getId()` in the line above. If this use shouldn't be there, how would we get the nodeId for the NoSuchNodeException that is thrown right in this block?

Reply:
@bleskes maybe you are familiar enough with this code to be able to judge whether node can be null here at all? From what I see so far, the nodes array is retrieved earlier in the method through `request.concreteNodes()`. Following the call hierarchy it is unlikely this can contain null references, but I don't think it is enforced anywhere. Then again, the reference in L179 hasn't led to NPEs since at least mid-2016 when it was added, so maybe its save to assume `node` cannot be null here.

Reply:
+1 remove the `node == null` check. IMO if we pass a null in there it's bug higher up in the chain which we should learn about and fix where it came from.